tool/templates/dart/struct.dart.jinja

```diff
@@ -75,13 +75,13 @@ final class {{type_name}} {
   bool operator ==(Object other) =>
       other is {{type_name}}
       {%- for field in fields %} &&
-      other.{{field.name}} == this.{{field.name}}
+      other.{{field.name}} == {% if field.name == "other" %}this.{% endif %}{{field.name}}
       {%- endfor %};
 
   @override
   int get hashCode => Object.hashAll([
       {%- for field in fields %}
-        this.{{field.name}},
```

Comment:
nit (nb): I prefer simplicity in the code generation and it's fine if that means being explicit more often

But no huge deal either way

Reply:
this produces `dart analyze` warnings and I'd like to be warning-free. It's either this or slapping allows on every single line